authors/apps/authentication/views.py

```diff
@@ -133,12 +147,24 @@ class VerifyAccountAPIView(APIView, JWTAuthentication):
 
     # function to retrieve user info from the token
     def get(self, request, token):
+        redirect_url = ''
+        cleantoken = token
+        if '$' in str(token):
+            redirect_url = str(base64.b64decode((token.rsplit('$', 1)[1])))
+            redirect_url = redirect_url[2:-1]
+            cleantoken = str((token.rsplit('$')[0]))
         try:
-            user, token = self.get_verification_credencials(request, token)
+
+            user, token = self.get_verification_credencials(
+                request, cleantoken)
 
             if not user.is_verified:
                 user.is_verified = True
                 user.save()
+
```

Comment:
Would you please use a helper function for return `HttpResponseRedirect(redirect_url)` to avoid code duplication